addons/web/static/src/views/view_button/view_button_hook.js

```diff
@@ -32,39 +32,52 @@ export async function executeButtonCallback(el, fct) {
 function undefinedAsTrue(val) {
     return typeof val === "undefined" || val;
 }
-export function useViewButtons(model, ref, options = {}) {
+
+/**
+ * @typedef {Object} ResParams
+ * @property {number} [resId]
+ * @property {string} [resModel]
+ * @property {number[]} [resIds]
+ * @property {Object} [context]
+ */
+
+/**
+ * @typedef {Object} Params
+ * @property {Function} [afterExecuteAction]
+ * @property {Function} [beforeExecuteAction]
+ * @property {Function} [getResParams]
+ * @property {() => ResParams} [reload]
+ */
+
+/**
+ * @param {{ readonly el: HTMLElement | null; }} ref
+ * @param {Params} [params={}]
+ */
+export function useViewButtons(ref, params = {}) {
     const action = useService("action");
     const dialog = useService("dialog");
     const comp = useComponent();
     const env = useEnv();
-    const beforeExecuteAction =
-        options.beforeExecuteAction ||
-        (() => {
-            return true;
-        });
-    const afterExecuteAction = options.afterExecuteAction || (() => {});
     useSubEnv({
         async onClickViewButton({ clickParams, getResParams, beforeExecute }) {
-            // const el = getEl();
             async function execute() {
                 let _continue = true;
                 if (beforeExecute) {
                     _continue = undefinedAsTrue(await beforeExecute());
                 }
 
-                _continue = _continue && undefinedAsTrue(await beforeExecuteAction(clickParams));
+                _continue =
+                    _continue && undefinedAsTrue(await params.beforeExecuteAction?.(clickParams));
                 if (!_continue) {
                     return;
                 }
                 const closeDialog =
                     (clickParams.close || clickParams.special) && env.dialogData?.close;
-                const params = getResParams();
-                const resId = params.resId;
-                const resIds = params.resIds || model.resIds;
+                const resParams = { ...params.getResParams?.(), ...getResParams() };
```

Comment:
a bit weird to be able to pass a getResParams function twice (once when calling the hook, and once when calling the handler)

Reply:
do you think we should make mandatory getResParams in onClickViewButton and have always a record in the ViewButton props?